open-apparel-registry/open-apparel-registry
# Add parent company to facility claim

## Overview

This PR adds a parent_company field to the FacilityClaim model as a FK to Contributor, and also enables selecting, displaying, and updating that field from the list of contributorOptions.

Connects #618 

## Demo
![Screen Shot 2019-06-22 at 10 44 55 AM](https://user-images.githubusercontent.com/4165523/59965401-739e1780-94db-11e9-872a-d6aff3671061.png)
![Screen Shot 2019-06-22 at 10 48 10 AM](https://user-images.githubusercontent.com/4165523/59965402-739e1780-94db-11e9-8364-b0d1c71f64c3.png)
![Screen Shot 2019-06-22 at 10 47 23 AM](https://user-images.githubusercontent.com/4165523/59965403-739e1780-94db-11e9-9ddc-a76914b012e6.png)
![Screen Shot 2019-06-22 at 10 45 08 AM](https://user-images.githubusercontent.com/4165523/59965404-739e1780-94db-11e9-9435-4589fa36aa5f.png)

## Notes

Adding the new field makes the number of items displayed in the dashboard claim details page uneven, which breaks using having the two text area fields sit adjacent at the bottom of the list. We can probably wait to fix this when and if it is an actual issue.

## Testing Instructions

- get this branch, then `resetdb` and `processfixtures` and `server`
- turn on claim a facility
- visit 6543 and verify that the sidebar select dropdown filters still work as they did before
- sign in as c2 and claim a facility. Verify that you can select a parent company from the select menu
- sign out and sign back in as c1, then visit /dashboard/claims and click on the facility claim
- verify that you see the parent company listed there and that it is linked to the correct profile
- approve the claim, then sign out, and sign back in as c1
- visit "My Facilities" and verify that you see the "Parent Company" field on the form and that you can change and save the "Parent Company" value
- visit the facility details page for the facility you've claimed and verify that you see the parent company linked in the claim info

## Checklist
